health/state.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package health
+
+import (
+	"time"
+)
+
+// State contains the health status of the backend, and is
+// provided to the client UI via LocalAPI through ipn.Notify.
+type State struct {
+	// Each key-value pair in Warnings represents a Warnable that is currently
+	// unhealthy. If a Warnable is healthy, it will not be present in this map.
+	// When a Warnable is unhealthy and becomes healthy, its key-value pair
+	// disappears in the next issued State. Observers should treat the absence of
+	// a WarnableCode in this map as an indication that the Warnable became healthy,
+	// and may use that to clear any notifications that were previously shown to the user.
+	Warnings map[WarnableCode]Representation
+}
+
+// Representation contains information to be shown to the user to inform them
+// that a Warnable is currently unhealthy.
+type Representation struct {
```

Comment:
I kinda hate this name :) .... all types describe a representation. And it's just very long. And "health.Representation" sounds broader than what this is, like it's describing all the health (like health.State).

Maybe `WarnableState` or `UnhealthyState` or `WarnMeta` or `UnhealthyMeta`?

Reply:
Done